### PR TITLE
Update Android Chat SDK APIs to follow JAVA

### PR DESCRIPTION
**Pull request by CHILIU-MSFT**
- Updated listMessages parameter from listChatMessagesOptions to listMessagesOptions 
- Updated addParticipants returns AddParticipantsResult
- Updated addParticipants & addParticipantsWithResponse parameter from AddParticipantsOptions option to Iterable<ChatParticipant> participants
- Minor BUG fix

### CODE REVIEW

File: sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/ChatThreadAsyncClient.java [hunk 1]
```diff
@@ -173,36 +172,36 @@ public String getChatThreadId() {
     /**
      * Adds participants to a thread. If participants already exist, no change occurs.
      *
-     * @param options Options for adding participants.
+     * @param participants Collection of participants to add.
```
Comment (vcolin7) [inline sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/ChatThreadAsyncClient.java:175]: Since there's a Java `Collection<T>` class, should we rephrase this to something like the following?

```suggestion
     * @param participants Participants to add.
```

File: sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/ChatThreadAsyncClient.java [hunk 2]
```diff
@@ -173,36 +172,36 @@ public String getChatThreadId() {
...
-                return null;
+                return response.getValue();
             });
     }
 
     /**
      * Adds participants to a thread. If participants already exist, no change occurs.
      *
-     * @param options Options for adding participants.
+     * @param participants Collection of participants to add.
```
Comment (vcolin7) [inline sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/ChatThreadAsyncClient.java:193]: Same comment as above about phrasing.

File: sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/ChatThreadAsyncClient.java [hunk 3]
```diff
@@ -264,16 +263,16 @@ public String getChatThreadId() {
...
```
Comment (vcolin7) [inline sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/ChatThreadAsyncClient.java:266]: Same comment as above about phrasing.

File: sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/ChatThreadClient.java [hunk 4]
```diff
@@ -109,25 +108,26 @@ public void updateTopic(String topic) {
...
-        block(this.client.addParticipants(options));
+    public AddChatParticipantsResult addParticipants(Iterable<ChatParticipant> participants) {
+        return block(this.client.addParticipants(participants));
     }
 
     /**
      * Adds participants to a thread. If participants already exist, no change occurs.
      *
-     * @param options options for adding participants.
+     * @param participants Collection of participants to add.
```
Comment (vcolin7) [inline sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/ChatThreadClient.java:122]: Same comment as above about phrasing.

File: sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/ChatThreadClient.java [hunk 5]
```diff
@@ -298,17 +298,17 @@ public ChatMessage getMessage(String chatMessageId) {
...
      *
-     * @param listChatMessagesOptions the list options.
+     * @param listMessagesOptions the list options.
      * @param context the context to associate with this operation.
      *
      * @return the list of thread messages.
      */
     @ServiceMethod(returns = ReturnType.COLLECTION)
-    public PagedIterable<ChatMessage> listMessages(ListChatMessagesOptions listChatMessagesOptions,
+    public PagedIterable<ChatMessage> listMessages(ListChatMessagesOptions listMessagesOptions,
```
Comment (vcolin7) [inline sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/ChatThreadClient.java:307]: Should we rename the type to `ListMessagesOptions` as well?

File: sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/implementation/converters/AddChatParticipantsOptionsConverter.java [hunk 6]
```diff
@@ -12,24 +11,24 @@
 
 /**
  * A converter between {@link com.azure.android.communication.chat.implementation.models.AddChatParticipantsOptions} and
- * {@link AddChatParticipantsOptions}.
+ * a collection of {@link ChatParticipant}.
```
Comment (vcolin7) [inline sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/implementation/converters/AddChatParticipantsOptionsConverter.java:14]: What about the following?

```suggestion
 * A {@link ChatParticipant} Iterable.
```

File: sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/implementation/converters/AddChatParticipantsOptionsConverter.java [hunk 7]
```diff
@@ -12,24 +11,24 @@
...
  */
 public final class AddChatParticipantsOptionsConverter {
     /**
-     * Maps from {AddChatThreadMembersOptions} to
+     * Maps from a collection of {@link ChatParticipant} to
```
Comment (vcolin7) [inline sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/implementation/converters/AddChatParticipantsOptionsConverter.java:18]:
```suggestion
     * Maps from a {@link ChatParticipant} Iterable to
```

File: sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/ChatThreadAsyncClient.java [hunk 8]
```diff
@@ -173,36 +172,36 @@ public String getChatThreadId() {
...
      *
      * @return the {@link CompletableFuture} that signals the operation result.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public CompletableFuture<Void> addParticipants(AddChatParticipantsOptions options) {
-        if (options == null) {
-            return CompletableFuture.failedFuture(new NullPointerException("'options' cannot be null."));
+    public CompletableFuture<AddChatParticipantsResult> addParticipants(Iterable<ChatParticipant> participants) {
```
Comment (vcolin7) [inline sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/ChatThreadAsyncClient.java:180]: What's the reason we went with `Iterable<ChatParticipant>` instead of something like  `List<ChatParticipant>`?

Comment (CHILIU-MSFT) [reply sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/ChatThreadAsyncClient.java:180]: This is from JAVA SDK. https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/ChatThreadAsyncClient.java#L156

File: sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/ChatThreadClient.java (hunk 5 above)
Comment (CHILIU-MSFT) [reply sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/ChatThreadClient.java:307]: Java also uses type name ListChatMessagesOptions. 

https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/communication/azure-communication-chat/src/main/java/com/azure/communication/chat/ChatThreadAsyncClient.java#L531

File: sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/ChatThreadAsyncClient.java (hunk 8 above)
Comment (anuchandy) [reply sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/ChatThreadAsyncClient.java:180]: It seems taking `Iterable` was added as part of a[ java-pr](https://github.com/Azure/azure-sdk-for-java/pull/19776) with title "Incoming Changes from Azure SDK review", could be that its a feedback from arch to take Iterable.

File: sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/ChatThreadClient.java (hunk 5 above)
Comment (anuchandy) [reply sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/ChatThreadClient.java:307]: Usually, as Victor pointed, the option bag name follows the pattern of `<simple-method-name>Options`, but it seems, java took the route of having "Chat" text in the Options model name e.g. "AddChatParticipantsOptions" not "AddParticipantsOptions", it could be that java used more specialized names to future proof, like in future if there is another client (not-chat) in the same module where it has API to add participants Or the case where Chat used with Call sdk which may support adding participants to a call.

**Review by anuchandy — APPROVED** [review]
LGTM, left my thoughts on the valid questions that Victor raised.

**Review by JianpingChen — APPROVED** [review]
